src/query.js

```diff
@@ -476,40 +476,38 @@ export default class Query {
    * @returns {Query} The query.
    */
   matches(field, regExp, options = {}) {
+    const flags = [];
+
     if (!isRegExp(regExp)) {
       regExp = new RegExp(regExp);
     }
 
-    if ((regExp.ignoreCase || options.ignoreCase) && options.ignoreCase !== false) {
-      throw new QueryError('ignoreCase flag is not supported.');
-    }
-
     if (regExp.source.indexOf('^') !== 0) {
-      throw new QueryError('regExp must have `^` at the beginning of the expression ' +
-        'to make it an anchored expression.');
+      throw new QueryError('regExp must have \'^\' at the beginning of the expression'
+        + ' to make it an anchored expression.');
     }
 
-    const flags = [];
+    if ((regExp.ignoreCase || options.ignoreCase) && options.ignoreCase !== false) {
```

Comment:
I still need to look into this. I didn't find it in the dev center but they linked the reference docs. The flag is there because you can make a reg ex case insensitive. `^/foo/i`